LEAF_Request_Portal/form.php

```diff
@@ -1342,48 +1342,131 @@ public function doSubmit($recordID)
      */
     public function getProgress($recordID)
     {
+        $returnValue = 0;
+
         $vars = array(':recordID' => (int)$recordID);
-        $tresRecord = $this->db->prepared_query('SELECT recordID, categoryID, count, submitted FROM records
+        //get all the catIDs associated with this record and whether the forms are enabled or submitted 
+        $resRecordInfoEachForm = $this->db->prepared_query('SELECT recordID, categoryID, `count`, submitted FROM records
                                                     LEFT JOIN category_count USING (recordID)
-                                                    WHERE recordID=:recordID', $vars);
-        $resRecord = array();
-        foreach ($tresRecord as $record)
-        {
-            if ($record['submitted'] > 0)
-            {
-                return 100;
-            }
-            $resRecord[strtolower($record['categoryID'])] = $record['count'];
+                                                    WHERE recordID=:recordID', $vars);                         
+        $isSubmitted = false;
+        foreach ($resRecordInfoEachForm as $request) {
+            if ($request['submitted'] > 0) {
+                $isSubmitted = true;
+                break;
+            } 
         }
+        if ($isSubmitted) {
+            $returnValue = 100;
 
-        $vars = array(':recordID' => (int)$recordID);
-        $resCompletedCount = $this->db->prepared_query('SELECT COUNT(*) FROM data LEFT JOIN indicators
-                                                            USING (indicatorID)
-                                                            WHERE recordID=:recordID
-                                                                AND indicators.required = 1
-        														AND indicators.disabled = 0
-                                                                AND data != ""', $vars);
+        } else {
 
-        $resCount = $this->db->prepared_query("SELECT categoryID, COUNT(*) FROM indicators WHERE required=1 AND disabled = 0 GROUP BY categoryID", array());
-        $countData = array();
-        $sum = 0;
-        foreach ($resCount as $cat)
-        {
-            $sum += $cat['COUNT(*)'] * (isset($resRecord[strtolower($cat['categoryID'])]) ? $resRecord[strtolower($cat['categoryID'])] : 0);
-        }
-        if ($sum == 0)
-        {
-            return 100;
-        }
+            //get indicatorID, data, and required for all completed questions for non-disabled indicators
+            $resCompleted = $this->db->prepared_query('SELECT indicatorID, `data`, `required` FROM `data` LEFT JOIN indicators
+                                                                USING (indicatorID)
+                                                                WHERE recordID=:recordID
+                                                                    AND indicators.disabled = 0
+                                                                    AND data != ""', $vars);
```

Comment:
```suggestion
                                                                    AND data != ""
                                                                    AND data != NULL', $vars);
```

Reply:
This shouldn't happen, but can add it just in case